src/test/java/de/hhu/propra/sharingplatform/controller/ItemControllerTest.java

```diff
@@ -0,0 +1,351 @@
+        user.setEmail("mail");
+        user.setAddress("address");
+        user.setName("name");
+        user.setBan(false);
+        user.setDeleted(false);
+        user.setId(1l);
+
+        Item item = new Item(user);
+        item.setAvailable(true);
+        item.setBail(1.0);
+        item.setDeleted(false);
+        item.setDescription("desc");
+        item.setLocation("loc");
+        item.setName("item");
+        //item.setOwner(user);
+        item.setPrice(2.0);
+        item.setId(3l);
+
+        Optional<Item> optI = Optional.of(item);
+        Optional<User> optU = Optional.of(user);
+
+        when(itemRepo.findById(3)).thenReturn(optI);
+        //when(userRepo.findByAccountName("accountname")).thenReturn(optU);
+        when(userService.fetchUserByAccountName("accountname")).thenReturn(user);
+        when(userService.fetchUserIdByAccountName("accountname")).thenReturn(1l);
+
+        mvc.perform(get("/item/details/3")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("Gegenstand bearbeiten")));
+    }
+
+    @Test
+    @WithMockUser("otheraccountname")
+    public void itemDetailsExistLoggedInNotOwner() throws Exception {
+        User user = new User();
+        user.setAccountName("accountname");
+        user.setPassword("password");
+        user.setEmail("mail");
+        user.setAddress("address");
+        user.setName("name");
+        user.setBan(false);
+        user.setDeleted(false);
+        user.setId(1l);
+
+        User user2 = new User();
+        user2.setAccountName("otheraccountname");
+        user2.setPassword("password");
+        user2.setEmail("mail");
+        user2.setAddress("address");
+        user2.setName("name2");
+        user2.setBan(false);
+        user2.setDeleted(false);
+        user2.setId(2l);
+
+        Item item = new Item(user);
+        item.setAvailable(true);
+        item.setBail(1.0);
+        item.setDeleted(false);
+        item.setDescription("desc");
+        item.setLocation("loc");
+        item.setName("item");
+        item.setOwner(user);
+        item.setPrice(2.0);
+        item.setId(3l);
+
+        Optional<Item> optI = Optional.of(item);
+        Optional<User> optU2 = Optional.of(user2);
+
+        when(itemRepo.findById(3)).thenReturn(optI);
+        when(userRepo.findByAccountName("otheraccountname")).thenReturn(optU2);
+
+        mvc.perform(get("/item/details/3")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("Leihe anfragen")));
+    }
+
+    @Test
+    @WithMockUser("accountname")
+    public void getNewItemLoggedIn() throws Exception {
+        mvc.perform(get("/item/new")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("Neuen Gegenstand anlegen")));

```

Comment:
maybe not the best case to compare because this needs to change to English.
We should use "item name" or other properties of the item